### PR TITLE
REPO-4589 Configure Mimetypes for custom renditions

### PR DESCRIPTION
**Pull request by alandavis**

### CODE REVIEW

**Comment by alandavis** [comment]
ConfigFileFinder and ConfigScheduler were moved from alfresco-repository

File: src/main/java/org/alfresco/repo/content/MimetypeMap.java [hunk 1]
```diff
@@ -346,17 +353,129 @@
...
+    private CronExpression cronExpression;
+
+    private CronExpression initialAndOnErrorCronExpression;
+
+    static class Data
+    {
+        private List<String> mimetypes = new ArrayList<String>(40);
 
-    private Map<String, String> extensionsByMimetype;
+        private Map<String, String> extensionsByMimetype = new HashMap<String, String>(59);;
```
Comment (Epurashu) [inline src/main/java/org/alfresco/repo/content/MimetypeMap.java]: Unnecessary semicolon

File: src/main/java/org/alfresco/repo/content/MimetypeMap.java [hunk 2]
```diff
@@ -422,6 +561,71 @@ public void init()
         PropertyCheck.mandatory(this, "configService", configService);
         PropertyCheck.mandatory(this, "contentCharsetFinder", contentCharsetFinder);
 
+        if (mimetypeJsonConfigDir != null || jsonObjectMapper != null || cronExpression != null || initialAndOnErrorCronExpression != null)
```
Comment (Epurashu) [inline src/main/java/org/alfresco/repo/content/MimetypeMap.java:565]: Do we really need this if-clause ?
From what I observe PropertyCheck.mandatory(Object target, String propertyName, Object value) verifies if an (value== null), I think we can change lines 564-570 to
```
PropertyCheck.mandatory(this, "jsonObjectMapper", jsonObjectMapper);
PropertyCheck.mandatory(this, "cronExpression", cronExpression);
PropertyCheck.mandatory(this, "initialAndOnErrorCronExpression", initialAndOnErrorCronExpression);
PropertyCheck.mandatory(this, "mimetypeJsonConfigDir", mimetypeJsonConfigDir);
```
I added mimetypeJsonConfigDir here as we verify that is not null when we read the JSON config (we can also remove that check - line 659)

**Review by Epurashu — APPROVED** [review]
lgtm :)

**Pull request merged**